flink-runtime/src/test/java/org/apache/flink/runtime/metrics/MetricRegistryImplTest.java

```diff
@@ -161,6 +161,26 @@ public void testReporterScheduling() throws Exception {
 		registry.shutdown().get();
 	}
 
+	@Test
+	public void testReporterIntervalParsingErrorDoesNotResultInPartialApplication() throws Exception {
+		TestReporter3.reportCount = 0;
+
+		MetricConfig config = new MetricConfig();
+		config.setProperty(ConfigConstants.METRICS_REPORTER_INTERVAL_SUFFIX, "1 UNICORN");
+
+		MetricRegistryImpl registry = new MetricRegistryImpl(
+			MetricRegistryConfiguration.defaultMetricRegistryConfiguration(),
+			Collections.singletonList(ReporterSetup.forReporter("test", config, new TestReporter3())));
+		try {
+			// in a prior implementation the time amount was applied even if the time unit was invalid
+			// in this case this would imply using 1 SECOND as the interval (seconds is the default)
+			Thread.sleep(2000);
+			Assert.assertEquals(0, TestReporter3.reportCount);
```

Comment:
Although, it is not very probable but possible test instability if the worker hangs for more than 10 sec.
In general, it would be nice to add a manual `Clock` to `ManuallyTriggeredScheduledExecutor` and inject it into `MetricRegistryImpl` constructor.